### PR TITLE
[#554] improvement(CI-Hive): improve Hadoop access permission

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?
 - set `hdfs` as HDFS superuser group in container
 - use `datastrato` as the Hive catalog Integration Test user instead of `root`

### Why are the changes needed?
we should not use the `root` user directly to access HDFS

Fix: #554 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
existing ITs

### CODE REVIEW

**Comment by mchades** [comment]
This PR, based on #620, adds owner validation for https://github.com/datastrato/gravitino/pull/620#pullrequestreview-1766767064.

@xunliu Please review it again, thanks!

**Review by xunliu — DISMISSED** [review]
LGTM

File: docs/docker-image-details.md [hunk 1]
```diff
@@ -68,6 +68,10 @@ You can use this kind of images to test the catalog of Apache Hive.
 
 Changelog
 
+- gravitino-ci-hive:0.1.7
+  - Download Mysql JDBC driver before building the Docker image
+  - Set `hdfs` as HDFS superuser group
+
 - gravitino-ci-hive:0.1.6
   - No start YARN when container startup
```
Comment (qqqttt123) [inline docs/docker-image-details.md:76]: Could you modify this grammar by the way?
`No start` -> `No starting`

File: docs/docker-image-details.md [hunk 2]
```diff
@@ -68,6 +68,10 @@ You can use this kind of images to test the catalog of Apache Hive.
 
 Changelog
 
+- gravitino-ci-hive:0.1.7
+  - Download Mysql JDBC driver before building the Docker image
```
Comment (qqqttt123) [inline docs/docker-image-details.md:72]: `Mysql` -> `MySQL`.

File: docs/docker-image-details.md (hunk 1 above)
Comment (mchades) [reply docs/docker-image-details.md]: OK, fixed

File: dev/docker/hive/hive-dependency.sh [hunk 3]
```diff
@@ -10,13 +10,17 @@ hive_dir="$(cd "${hive_dir}">/dev/null; pwd)"
...
+MYSQL_JDBC_DRIVER_VERSION="8.0.15"
 
 HADOOP_PACKAGE_NAME="hadoop-${HADOOP_VERSION}.tar.gz" # Must export this variable for Dockerfile
 HADOOP_DOWNLOAD_URL="http://archive.apache.org/dist/hadoop/core/hadoop-${HADOOP_VERSION}/${HADOOP_PACKAGE_NAME}"
 
 HIVE_PACKAGE_NAME="apache-hive-${HIVE_VERSION}-bin.tar.gz" # Must export this variable for Dockerfile
 HIVE_DOWNLOAD_URL="https://archive.apache.org/dist/hive/hive-${HIVE_VERSION}/${HIVE_PACKAGE_NAME}"
 
+JDBC_DIVER_PACKAGE_NAME="mysql-connector-java-${MYSQL_JDBC_DRIVER_VERSION}.tar.gz" # Must export this variable for Dockerfile
+JDBC_DIVER_DOWNLOAD_URL="https://downloads.mysql.com/archives/get/p/3/file/${JDBC_DIVER_PACKAGE_NAME}"
```
Comment (qqqttt123) [inline dev/docker/hive/hive-dependency.sh:22]: Why do you use this version? We used another version in JDBC integration tests and playground.
https://repo1.maven.org/maven2/mysql/mysql-connector-java/8.0.27/mysql-connector-java-8.0.27.jar

**Comment by qqqttt123** [comment]
Will you raise a follow-up pr to modify the Docker image which integration-test uses and the document?

File: dev/docker/hive/hive-dependency.sh (hunk 3 above)
Comment (mchades) [reply dev/docker/hive/hive-dependency.sh:22]: We have been using this version from the beginning(see following ref) in Hive-CI-Docker, but the JDBC integration tests and the playground introduced later are inconsistent with the previous version.

https://github.com/datastrato/gravitino/blob/dbdc1c9a6cbf31cff9d56c2e93a8bf8c1a9b13cd/dev/docker/hive/Dockerfile#L131-L132

File: dev/docker/hive/hive-dependency.sh [hunk 4]
```diff
@@ -29,3 +33,7 @@ fi
 if [ ! -f "${hive_dir}/packages/${HIVE_PACKAGE_NAME}" ]; then
   curl -s -o "${hive_dir}/packages/${HIVE_PACKAGE_NAME}" ${HIVE_DOWNLOAD_URL}
 fi
+
+if [ ! -f "${hive_dir}/packages/${JDBC_DIVER_PACKAGE_NAME}" ]; then
+  curl -L -s -o "${hive_dir}/packages/${JDBC_DIVER_PACKAGE_NAME}" ${JDBC_DIVER_DOWNLOAD_URL}
+fi
```
Comment (qqqttt123) [inline dev/docker/hive/hive-dependency.sh:39]: A new line.

File: dev/docker/hive/hive-dependency.sh (hunk 3 above)
Comment (qqqttt123) [reply dev/docker/hive/hive-dependency.sh:22]: OK, we don't need to keep consistent in this pr. I guess we can change them to a consistent version in later pr.

**Comment by mchades** [comment]
> Will you raise a follow-up pr to modify the Docker image which integration-test uses and the document?

The Docker image `datastrato/gravitino-ci-hive:0.1.7` was pushed to the [DockerHub](https://hub.docker.com/r/datastrato/gravitino-ci-hive/tags). The integration test image that the PR currently uses and passes is the latest.

The document also modified in this PR(see file `docs/docker-image-details.md`)

File: integration-test/build.gradle.kts [hunk 5]
```diff
@@ -265,14 +265,13 @@ tasks.test {
...
-      // TODO: use hive user instead after we fix the permission issue #554
-      environment("HADOOP_USER_NAME", "root")
+      environment("HADOOP_USER_NAME", "datastrato")
       environment("HADOOP_HOME", "/tmp")
       environment("PROJECT_VERSION", version)
       environment("TRINO_CONF_DIR", buildDir.path + "/trino-conf")
 
       // Gravitino CI Docker image
-      environment("GRAVITINO_CI_HIVE_DOCKER_IMAGE", "datastrato/gravitino-ci-hive:0.1.6")
+      environment("GRAVITINO_CI_HIVE_DOCKER_IMAGE", "datastrato/gravitino-ci-hive:0.1.7")
```
[the remark this answers is not in the document]
Comment (mchades) [reply integration-test/build.gradle.kts:274]: Yep

[the remark this answers is not in the document]
Comment (qqqttt123) [reply integration-test/build.gradle.kts:274]: OK.

File: dev/docker/hive/hive-dependency.sh (hunk 4 above)
Comment (mchades) [reply dev/docker/hive/hive-dependency.sh]: added

**Review by qqqttt123 — APPROVED** [review]
LGTM.

**Comment by jerryshao** [comment]
Do you also need to ship a new hive image for users? cc @qqqttt123

**Comment by qqqttt123** [comment]
> Do you also need to ship a new hive image for users? cc @qqqttt123

No. This image is used by developers.

**Pull request merged**